config/routes.rb

```diff
@@ -1,5 +1,9 @@
 Rails.application.routes.draw do
   resources 'messages', :only => [:index, :new, :create, :show, :destroy]
+  
+  get 'recipients/:id' => 'recipients#show'
```

Comment:
ok, will give this: `resources :recipients, only: [:show]` a try. 

Suppose I could use default routes, though the authentication post would be a bit of a misnomer.

GET: index <-- auth form or redirect to show if auth is not required
GET: show <-- show message
POST: create <-- authenticate